versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractMergeTransplant.java

```diff
@@ -148,26 +167,31 @@ void apply(
         throws Exception;
   }
 
-  private Hash[] mergeTransplant(MergeOrTransplant mergeOrTransplant) throws Exception {
+  private Hash[] mergeTransplant(int numCommits, MergeOrTransplant mergeOrTransplant)
+      throws Exception {
     BranchName main = BranchName.of("main");
     BranchName branch = BranchName.of("branch");
     BranchName conflict = BranchName.of("conflict");
 
     databaseAdapter.create(branch, databaseAdapter.hashOnReference(main, Optional.empty()));
 
-    Hash[] commits = new Hash[3];
+    Hash[] commits = new Hash[numCommits];
     for (int i = 0; i < commits.length; i++) {
       ImmutableCommitAttempt.Builder commit =
           ImmutableCommitAttempt.builder()
               .commitToBranch(branch)
               .commitMetaSerialized(ByteString.copyFromUtf8("commit " + i));
-      for (int k = 0; k < 3; k++) {
-        commit.addPuts(
-            KeyWithBytes.of(
-                Key.of("key", Integer.toString(k)),
-                ContentId.of("C" + k),
-                (byte) 0,
-                ByteString.copyFromUtf8("value " + i + " for " + k)));
+      if (i == commits.length - 1) {
+        commit.addDeletes(Key.of("key", Integer.toString(1)));
```

Comment:
Should also have deletes for previously added keys - also in commits "in the middle". We really want to be sure that the change works as expected.

Reply:
deleting in the middle now.